keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
-  // to minimize the possibility of collisions. It MUST be 63
-  // characters or less, beginning and ending with an alphanumeric
-  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
-  // alphanumerics between. This field is REQUIRED.
-  string name = 1;
+```yaml
+BucketClass {
+  TypeMeta
+  ObjectMeta
+
+  // Provisioner is the name of driver associated with this bucket
+  Provisioner string
+
+  // Protocol is the data API this bucket is expected to adhere to.
+  // The possible values for protocol are:
+  // -  S3: Indicates Amazon S3 protocol
+  // -  Azure: Indicates Microsoft Azure BlobStore protocol
+  // -  GCS: Indicates Google Cloud Storage protocol
+  Protocol Protocol
+
+  // DeletionPolicy is used to specify how COSI should handle deletion of this
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  DeletionPolicy DeletionPolicy
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisonerCreateBucket
+## BucketAccess
 
-This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner. The call to *ProvisonerCreateBucket* MUST be idempotent.
+A resource to access a Bucket. BucketAccess is namespace-scoped
 
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketClaimSpec {
+    // BucketClaimName is the name of the BucketClaim.
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketClaimName string
+
+    // BucketName is the name of the Bucket for which
+    // credetials need to be generated
+    // Exactly one of BucketClaimName or BucketName must be set.
+    // +optional
+    BucketName string
+
+    // BucketAccessClassName is the name of the BucketAccessClass
+    BucketAccessClassName string
+
+    // CredentialsSecretName is the name of the secret that COSI should populate
+    // with the credentials. If a secret by this name already exists, then it is
+    // assumed that credentials have already been generated. It is not overridden.
+    CredentialsSecretName string
+    
+    // ServiceAccountName is the name of the serviceAccount that COSI will map
+    // to the OSP service account when IAM styled authentication is specified
+    CredentialsSecretName string
+  }
+
+  Status BucketAccessStatus {
+    // AccessGranted indicates the successful grant of privileges to access the bucket
+    AccessGranted bool
+    
+    // AccountID is the unique ID for the account in the OSP
+    AccountID string
```

Comment:
Does the user use this account id for anything?

Reply:
The user does not, however, it is used by COSI for revoking access.

Reply:
Can you specify when the field will be populated?

Reply:
I'll add it to the document as well. The field will be populated by the COSI sidecar once access has been successfully granted.